components/rule-mgt/org.wso2.carbon.identity.rule.management/src/main/java/org/wso2/carbon/identity/rule/management/dao/impl/RuleManagementDAOImpl.java

```diff
@@ -0,0 +1,316 @@
+                            statement.setBoolean(RuleSQLConstants.Column.IS_ACTIVE, true);
+                            statement.setString(RuleSQLConstants.Column.RULE_EXTERNAL_ID, ruleId);
+                            statement.setInt(RuleSQLConstants.Column.TENANT_ID, tenantId);
+                        });
+                return null;
+            });
+        } catch (TransactionException e) {
+            throw new RuleManagementServerException("Error while activating the rule in the system.", e);
+        }
+    }
+
+    /**
+     * This method will deactivate the Rule in the database.
+     *
+     * @param ruleId   Rule ID
+     * @param tenantId Tenant ID
+     * @throws RuleManagementException If an error occurs while deactivating the rule in the database.
+     */
+    @Override
+    public void deactivateRule(String ruleId, int tenantId) throws RuleManagementException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            jdbcTemplate.withTransaction(template -> {
+                jdbcTemplate.executeUpdate(RuleSQLConstants.Query.CHANGE_RULE_STATUS,
+                        statement -> {
+                            statement.setBoolean(RuleSQLConstants.Column.IS_ACTIVE, false);
+                            statement.setString(RuleSQLConstants.Column.RULE_EXTERNAL_ID, ruleId);
+                            statement.setInt(RuleSQLConstants.Column.TENANT_ID, tenantId);
+                        });
+                return null;
+            });
+        } catch (TransactionException e) {
+            throw new RuleManagementServerException("Error while deactivating the rule in the system.", e);
+        }
+    }
+
+    private int addRuleToDB(Rule rule, int tenantId)
+            throws TransactionException, IOException, RuleManagementServerException {
+
+        InputStream ruleJsonAsInputStream = convertRuleToJson(rule);
+        int ruleJsonStreamLength = ruleJsonAsInputStream.available();
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        AtomicInteger internalId = new AtomicInteger();
+        jdbcTemplate.withTransaction(template -> {
+            internalId.set(template.executeInsert(RuleSQLConstants.Query.ADD_RULE,
+                    statement -> {
+                        statement.setString(RuleSQLConstants.Column.RULE_EXTERNAL_ID, rule.getId());
+                        statement.setBinaryStream(RuleSQLConstants.Column.RULE, ruleJsonAsInputStream,
+                                ruleJsonStreamLength);
+                        statement.setBoolean(RuleSQLConstants.Column.IS_ACTIVE, rule.isActive());
+                        statement.setInt(RuleSQLConstants.Column.TENANT_ID, tenantId);
+                        statement.setString(RuleSQLConstants.Column.VERSION, V1);
+                    }, rule, true));
+            return null;
+        });
+
+        return internalId.get();
```

Comment:
Some db types doesn't support returning the auto generated id. In such cases we may need to check whether the value of `internalId` is zero and then use a different way to fetch the id (Eg: getRuleByUUID)

Reply:
Fixed